anta/catalog.py

```diff
@@ -283,14 +287,17 @@ def from_list(data: ListAntaTestTuples) -> AntaCatalog:
         return AntaCatalog(tests)
 
     def get_tests_by_tags(self, tags: list[str], strict: bool = False) -> list[AntaTestDefinition]:
-        """
-        Return all the tests that have matching tags in their input filters.
+        """Return all the tests that have matching tags in their input filters.
+
         If strict=True, returns only tests that match all the tags provided as input.
         If strict=False, return all the tests that match at least one tag provided as input.
         """
         result: list[AntaTestDefinition] = []
         for test in self.tests:
             if test.inputs.filters and (f := test.inputs.filters.tags):
-                if (strict and all(t in tags for t in f)) or (not strict and any(t in tags for t in f)):
+                if strict:
+                    if all(t in tags for t in f):
+                        result.append(test)
+                elif any(t in tags for t in f):
```

Comment:
not convinced :p

Reply:
Pylint was not happy: https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/too-many-boolean-expressions.html

Reply:
lets make pylint happy without this terrible block then :p

Reply:
This is actually more readable than before, but there are multiple refactoring possible.